lib/tmp.js

```diff
@@ -220,6 +220,14 @@ function _createTmpFile(options, callback) {
     fs.open(name, CREATE_FLAGS, opts.mode || FILE_MODE, function _fileCreated(err, fd) {
       if (err) return cb(err);
 
+      if (opts.discardDescriptor) {
+        return fs.close(fd, function _invokeCallback(err) {
+          cb(null, name, undefined, _prepareTmpFileRemoveCallback(name, -1, opts));
```

Comment:
I'd considered that and figured the chance of a close error at this point (where nobody else can see the file) is tiny, and besides the file was successfully created so the call should succeed.

An alternative perspective which is somewhat compelling is to return the error, but then we also must unlink the file.  If that also errors, we have a problem because we can only return one error.  In that case the unlink error should be returned instead of the close error, as it's more important for the programmer to know that an orphan file might be present than that a descriptor got leaked.

I've added a separate commit with this behavior so we can see what it looks like, but it can't be tested since I can't force a failure in the close operation without completely reworking `tmp`'s unit test infrastructure.